packages/sanity/src/desk/comments/src/context/setup/CommentsSetupProvider.tsx

```diff
@@ -0,0 +1,128 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react'
+import {SanityClient} from '@sanity/client'
+import {CommentPostPayload} from '../../types'
+import {CommentsSetupContext} from './CommentsSetupContext'
+import {CommentsSetupContextValue} from './types'
+import {useWorkspace, useClient, DEFAULT_STUDIO_CLIENT_OPTIONS} from 'sanity'
+
+interface CommentsClientProviderProps {
+  children: React.ReactNode
+}
+
+/**
+ * @beta
+ * @hidden
+ */
+export function CommentsSetupProvider(props: CommentsClientProviderProps) {
+  const {children} = props
+  const {dataset, projectId} = useWorkspace()
+  const originalClient = useClient(DEFAULT_STUDIO_CLIENT_OPTIONS)
+  const [addonDatasetClient, setAddonDatasetClient] = useState<SanityClient | null>(null)
+  const [isRunningSetup, setIsRunningSetup] = useState<boolean>(false)
+
+  const getAddonDatasetName = useCallback(async (): Promise<string | undefined> => {
+    const res = await originalClient.withConfig({apiVersion: 'vX'}).request({
+      uri: `/projects/${projectId}/datasets?datasetProfile=comments&addonFor=${dataset}`,
```

Comment:
Should there be a  tag added to this request too? e.g. `sanity.studio.(...)` (you're in a better position to judge what the particular tag should be)

Reply:
Good question 🤔 I am not sure – will check with the team!

Reply:
Yes, this should be the case. Fixed! Great catch 👍